lib/orthoses/active_record/scope.rb

```diff
@@ -58,6 +58,16 @@ def parameters_to_type(parameters)
         end
         "(#{res.join(", ")})#{block}"
       end
+
+      def parameters(body)
+        if body.is_a?(Proc)
+          body.parameters
+        elsif body.respond_to?(:call)
+          body.method(:call).parameters
+        else
+          raise "unexpected: #{body} passed to scope"
+        end
```

Comment:
In Rails, it seems to branch based on the presence of `#to_proc`.
Also, since the `#call` presence is checked by the `#scope`, it seems unnecessary to check for `#call` here.
Therefore, how about making the following correction?

```suggestion
        if body.respond_to?(:to_proc)
          body.to_proc.parameters
        else
          body.method(:call).parameters
        end
```

ref:
https://github.com/rails/rails/blob/8c82595b8c862eb087615a14f61ded528274eeae/activerecord/lib/active_record/scoping/named.rb#L173-L185

Reply:
@ksss 

Thank you for the review. I agree your suggestion, and apply it.